### PR TITLE
[DG] Add Overview, Sequence Diagram, and Use Case for Statistic

### PR DESCRIPTION
**Pull request by Eclmist**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -1276,6 +1276,54 @@ will be filtered.
 
 The command used to filter claims is implemented the same way as `SessionFilterCommand`.
 
+=== Statistic Report Generation
+
+==== Description
+
+The *Statistics Report Window* can be generated and displayed using the `report` command.
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: Shouldnt this just be StatisticWindow since that's what the file is called?

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -1276,6 +1276,54 @@ will be filtered.
...
+* The number of hours of each type of completed sessions
+* A breakdown of your student’s ratings
+
+A module code can be specified such that the generated report will only include data from a specific module.
+
+==== Implementation
+
+This section describes the implementation of the `report` command.
+
+The following Sequence Diagram shows the interactions between the `UI` and the `Logic` components of TA-Tracker,
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: Need to italicise _Sequence Diagram_ so that its consistent with the rest :)

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -1276,6 +1276,54 @@ will be filtered.
...
+with the command `report CS3247`.
+
+. The user command is first read by `MainWindow`, through JavaFX.
+`MainWindow` passes the command as a `String` to the `LogicManager` to be processed.
+
+. `LogicManager` sends the command to `TaTrackerParser` for the command to be parsed.
+
+. The `TaTrackerParser` processes the first word in the command, and identifies it as a `ShowStatisticCommand`.
+Hence, `TaTrackerParser` creates a `ShowStatisticCommandParser` object and passes the command argument `CS3247`
+to the `ShowStatisticCommandParser` object.
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: you could probably break this whole chunk into 2 separate points

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -1276,6 +1276,54 @@ will be filtered.
...
+
+. The `ShowStatisticCommandParser` stores the target module, `CS3247`, in a `ShowStatisticCommand` object and
+this command object is returned all the way back to the `LogicManager`.
+
+. `LogicManager` executes the `ShowStatisticCommand`, which creates and return a `StatiscCommandResult`. This command
+result is returned by `LogicManager` to `MainWindow`
+
+. `MainWindow` detects that the the command result is of type `StatisticCommandResult`, and prepares the `StatisticWindow`
+by creating a `Statistic` object that retreives data necessary for generating the report, from `ReadOnlyTaTracker`. These data is then processed further by `Statistic`. This includes computing the total number of sessions per session type and sorting the students by rating.
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: this chunk here too

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -1850,6 +1898,34 @@ Use case ends.
...
+=== Statistic Report Window
+
+[discrete]
+==== Use case: UC23 - Displaying a statistic report
+
+*MSS*
+
+1. User requests to generate and display a statistic report.
+2. TA-Tracker retreives the list of completed sessions, and students.
+3. TA-Tracker processes the data retreived.
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: I think you should combine these 2 so it looks more ping-pongy, also I feel like just here you should mention its the full list for all the modules or smth

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -1850,6 +1898,34 @@ Use case ends.
...
+4. TA-Tracker opens a Statistic window and displays the processed data to the user.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The user did not specify a specific module code.
```
Comment (fatin99) [inline docs/DeveloperGuide.adoc]: And then this extension should be "the user specified a module code"
1a1. TA-Tracker only retrieve data from the sessions and students associated with the module
Use case ends

or whichever way you want

Comment (Eclmist) [reply docs/DeveloperGuide.adoc]: Ah good point

Comment (Eclmist) [reply docs/DeveloperGuide.adoc]: Would I then still need the subcase for if the module code is not valid?

**Review by fatin99 — APPROVED** [review]
Looks Good

**Pull request merged**